### ISSUE TITLE
Missing `<mo rspace="thickmathspace">⁢</mo>` spacing before a semantic Unit when the Unit is a composed unit

### ISSUE THREAD

**Issue by opoudjis** [plurimath/plurimath#314]
From https://github.com/metanorma/metanorma-bipm/issues/442

There is a difference in rendering between 2021 and 2024 for formulas: there was more space between numbers in a block formula.

The difference between the two renderings was that in 2021, we had more mrow wrappers in the MathML. We *may* need to resume putting mrow wrappers into MathML, but I'm not enthusiastic about it.

This is an issue specific to PDF rendering, so @Intelligent2013 needs to weigh in here. In HTML, the new rendering with fewer mrow wrappers initially renders without spaces, 

![formula-1-html-loading](https://github.com/user-attachments/assets/7e281caf-0513-422a-a014-8f8adbe7038e)

and then refreshes to insert spaces correctly:

![formula-1-html](https://github.com/user-attachments/assets/bd92fe00-97e2-4bf8-a121-55b1ca381e0a)

In Jan 2021 we generated in PDF:

![formula-1-before](https://github.com/user-attachments/assets/452f0fdf-cda2-4f42-8943-64ac30a52b57)

We are now generating in PDF:

![formula-1](https://github.com/user-attachments/assets/5b8975ed-a15d-4c25-8ee0-89c78d7b19a9)

The Presentation XML before was:

````
<formula id="_150eca79-1f6d-44d1-97b3-19258628cdbd" unnumbered="true">
<stem type="MathML"><math xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mtext> </mtext><mstyle mathvariant="normal"><mrow><mi>c</mi><mi>d</mi></mrow></mstyle><mo>=</mo><mfenced open="(" close=")"><mrow><mfrac><mrow><msub><mrow><mstyle mathvariant="italic"><mi>K</mi></mstyle></mrow><mrow><mtext>cd</mtext></mrow></msub></mrow><mrow><mn>683</mn></mrow></mfrac></mrow></mfenced><mtext> </mtext><mstyle mathvariant="normal"><mrow><mi>k</mi><mi>g</mi></mrow></mstyle><mtext> </mtext><msup><mrow><mstyle mathvariant="normal"><mi>m</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msup><mtext> </mtext><msup><mrow><mstyle mathvariant="normal"><mi>s</mi></mstyle></mrow><mrow><mrow><mo>−</mo><mn>3</mn></mrow></mrow></msup><mtext> </mtext><msup><mrow><mstyle mathvariant="normal"><mrow><mi>s</mi><mi>r</mi></mrow></mstyle></mrow><mrow><mrow><mo>−</mo><mn>1</mn></mrow></mrow></msup></math></stem>
</formula>
````

It is now

````
<formula id="_ef371af9-55b7-e5c8-413b-3736db69011e_si-brochure-en" unnumbered="true"><stem block="true" type="MathML"><math xmlns="http://www.w3.org/1998/Math/MathML">
  <mstyle displaystyle="true">
    <mn>1</mn>
    <mo rspace="thickmathspace">⁢</mo>
    <mrow xref="U_NISTu7">
      <mstyle mathvariant="normal">
        <mi>cd</mi>
      </mstyle>
      
      
      
    </mrow>
    <mo>=</mo>
    <mo rspace="thickmathspace">⁢</mo>
    <mrow>
      <mo>(</mo>
      <mfrac>
        <msub>
          <mstyle mathvariant="italic">
            <mi>K</mi>
          </mstyle>
          <mtext>cd</mtext>
        </msub>
        <mn>683</mn>
      </mfrac>
      <mo>)</mo>
    </mrow>
    <mrow xref="U_kg.m2.s-3.sr-1">
      <mstyle mathvariant="normal">
        <mi>kg</mi>
      </mstyle>
      <mo>⋅</mo>
      <msup>
        <mstyle mathvariant="normal">
          <mi>m</mi>
        </mstyle>
        <mn>2</mn>
      </msup>
      <mo>⋅</mo>
      <msup>
        <mstyle mathvariant="normal">
          <mi>s</mi>
        </mstyle>
        <mrow>
          <mo>−</mo>
          <mn>3</mn>
        </mrow>
      </msup>
      <mo>⋅</mo>
      <msup>
        <mstyle mathvariant="normal">
          <mi>sr</mi>
        </mstyle>
        <mrow>
          <mo>−</mo>
          <mn>1</mn>
        </mrow>
      </msup>
      
      
      
    </mrow>
  </mstyle>
</math><asciimath>1 "unitsml(cd)" = (ii(K)_("cd")/683) "unitsml(kg*m^2*s^(-3)*sr^(-1))"</asciimath></stem></formula>
````

Before:

````
<msup>
  <mrow>
    <mstyle mathvariant="normal">
      <mi>s</mi>
    </mstyle>
  </mrow>
  <mrow>
    <mrow>
      <mo>−</mo><mn>3</mn>
    </mrow>
  </mrow>
</msup>
````

After:

````
  <msup>
        <mstyle mathvariant="normal">
          <mi>s</mi>
        </mstyle>
        <mrow>
          <mo>−</mo>
          <mn>3</mn>
        </mrow>
      </msup>
  ````

It seems that the before renders more nicely in PDF than the after. I leave that to you two to sort out.

In relation to: https://github.com/metanorma/bipm-si-brochure/issues/240

**Comment by ronaldtse** [comment]
After some understanding, this is a bug originally implemented per:
* https://github.com/plurimath/plurimath/pull/221#discussion_r1515925979

In the original issue, the request was to insert whitespace *before* every (UnitsML) unit. Currently, the whitespace is present when the unit is a regular (singular) unit, but when it is a composed unit, the space is not rendered.

This is a bug that needs to be addressed ASAP.

**Comment by ronaldtse** [comment]
Released in 0.8.19.
